R/make-array.R

```diff
@@ -80,6 +80,26 @@ makeMR <- function (subvariables, name, selections, ...) {
     return(vardef)
 }
 
+#' Array builder
+#'
+#' Launch array builder gadget
+#'
+#' Categorical Array and Multiple Response variables can be difficult to
+#' construct without being able to investigate the available variables, and
+#' their categories. This shiny gadget lets you select subvariables from the
+#' dataset list, and ensures that those variables have consistent categories. To
+#' use the gadget you must have at least one CrunchDataset loaded into the global
+#' environment.
+#'
+#' @return a valid call to `makeArray()` or `makeMR()`
+#' @export
+makeArrayGadget <- function(){
+    if ("makeArrayGadget" %in% ls(envir = asNamespace("crunchy"))) {
```

Comment:
If you wanted to test this function, or at least the `stop` side of the `if` (and I'm not saying you need to), you could factor out this check to something like `has_function <- function (fun, pkg)`, and then mock `has_function` to return `FALSE`.

Reply:
Oh that's a good strategy. I think for this case it's not a test that adds a whole lot of value so I'd prefer to not include it. But still a good idea to add a function to check if a package has the right function.